Mathlib/Data/List/Perm/Lattice.lean

```diff
@@ -28,6 +29,23 @@ variable {α : Type*}
 
 open Perm (swap)
 
+lemma Nodup.splits {l : List α} : l.splits.Nodup :=
```

Comment:
We usually reserve names like `Nodup.f` for lemmas of the form `l.Nodup -> l.f.Nodup`. Your lemmas are not of this form, so I would go with `nodup_splits` instead. Same for the other lemmas in this file.